game-core/src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java

```diff
@@ -1652,14 +1652,13 @@ private boolean getCanAllUnitsWithRequiresUnitsBePlacedCorrectly(
     if (to == null) {
       return new ArrayList<>();
     }
-    final Collection<Unit> unitsInTo = to.getUnits();
     final Collection<Unit> unitsPlacedAlready = getAlreadyProduced(to);
     if (Matches.territoryIsWater().test(to)) {
       for (final Territory current : getAllProducers(to, player, null, true)) {
         unitsPlacedAlready.addAll(getAlreadyProduced(current));
       }
     }
-    final Collection<Unit> unitsAtStartOfTurnInTo = new ArrayList<>(unitsInTo);
+    final Collection<Unit> unitsAtStartOfTurnInTo = new ArrayList<>(to.getUnits());
```

Comment:
Would it be simpler to have the copy be done once and consistently in the source method?


EG: UnitCollection.java
```
  default Collection<Unit> getUnits() {
    return getUnitCollection().getUnits();
  }
```

Reply:
The whole point is to not copy.

The places that are changing are the few places that still need to copy, but those are the ones we don't care about. The point is all the other places are now *not* copying - and those are the ones that were contributing to the run time.

Reply:
Sorry, mistyped, meant to suggest we do the unmodifiable list copy here.

It's not super surprising we see some issues when copying, the usages of this method all seem to assume a constant time operation and invoke the method all over the place, even if the list is not expected to change.

Reply:
I'm still not following your suggestions.

"meant to suggest we do the unmodifiable list copy here"

The next line modifies the list copy, so here it needs be modifiable.